app/Services/SC/ScCoreService.cs

```diff
@@ -109,15 +109,21 @@ public async Task<ScCaseSearchViewModel> GetSearchResults(ScCaseSearchViewModel
             return newModel;
         }
 
-        public void SaveSearchForm(ScCaseSearchViewModel model)
+        public async Task SaveSearchForm(ScCaseSearchViewModel model)
         {
             var bookingInfo = _session.ScBookingInfo;
 
+            // we need to do a second API call to get the selectedCourtFile because
+            // if we passed it with hidden fields then there would be a security
+            // vulnerability where an attacker could modify fairUseSort to get into
+            // an earlier lottery round
```

Comment:
I think this comment might be explaining too much, but at the same time I wanted to make sure future devs are aware that this is a really important field to safeguard.

Reply:
Looks good to me, and fairly concise. It's pretty rare to ever see "too many" comments. Sometimes it feels like a lot, but for future developers reading them, the more the merrier 👍